feature: Add /api/v1/users/_login endpoint

3dxp opened the pull request:
Create the endpoint for getting a JWT when providing
a username and password.

#51

zxkevin committed:
feat: Add password & PUT endpoint to Users

Added new migrations to update Users table with password column
Seed file added to update pre-existing seeds with a password.
POST endpoint for Users now requires a password.
PUT endpoint for {userID} added for update users functionality.
Bcrypt is used to salt and hash password for secure password storage.

3dxp committed:
feature: Add /api/v1/users/_login endpoint

Create the endpoint for getting a JWT when providing
a username and password.

#51

3dxp committed:
feature: Reslove Merge Conflicts

Merge changes from upstream into the current working
branch.

3dxp commented:
This PR is dependent on work in #90, so please review that before this.

acampb65 commented on packages/hackmcx-api/src/utils/validation.js:
```diff
@@ -0,0 +1,5 @@
+const whitespaceRegex = /^\s*$/.compile();
```
```suggestion
const whitespaceRegex = new RegExp("/^\s*$/");
```
Apparently `RegExp.compile()` is deprecated and actually didn't work properly on my machine. If I'm not mistaken using the constructor instead gives us the same functionality and did end up working on my end at least.

acampb65 replied:
`whitespaceRegex = /^\s*$/` works as well. 
`whitespaceRegex = new RegExp("/^\s*$/")` would be redundant, I think.

acampb65 committed:
fix: Regular Expression Compile

compile() is deprecated and unsupported. Replaced with RegExp constructor for compatibility.

zxkevin committed:
fix: Update old seed, PUT inserts and updates

Updated the old seed data so that a password field exists
so that seeding will not break
Updated the PUT endpoint so that it will create new users if not
already in database and update users if in database

#87 #51

3dxp committed:
Merge branch 'main' into enhancement/3dxp/login

# Conflicts:
#	packages/hackmcx-api/package-lock.json
#	packages/hackmcx-api/package.json
#	packages/hackmcx-api/src/controllers/users.controller.js
#	packages/hackmcx-api/src/routers/v1/users.router.js

acampb65 commented on packages/hackmcx-api/src/utils/validation.js:
```diff
@@ -0,0 +1,5 @@
+const whitespaceRegex = /^\s*$/.compile();
```
```suggestion
const whitespaceRegex = new RegExp("/^\s*$/");
```

acampb65 committed:
Update packages/hackmcx-api/src/utils/validation.js

compile() is deprecated. Used constructor instead.
